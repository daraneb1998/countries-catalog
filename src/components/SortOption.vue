<script setup lang="ts">
import { useSearchStore } from "@/stores/search";
import { SortType } from "@/types";

import { useLoading } from "vue-loading-overlay";

const $loading = useLoading({
  isFullPage: true,
});

const searchStore = useSearchStore();
const onAscendingSort = () => {
  const loader = $loading.show();
  searchStore.updateSortType(SortType.ASCENDING);
  loader.hide();
};
const onDescendingSort = () => {
  const loader = $loading.show();
  searchStore.updateSortType(SortType.DESCENDING);
  loader.hide();
};
</script>

<template>
  <div class="flex gap-x-2 flex-wrap sticky">
    <button
      disabled
      class="border border-blue-500 text-gray-500 px-4 bg-slate-200 transition-colors"
    >
      SORT NAME
    </button>
    <button
      class="border border-blue-500 hover:text-white px-4 hover:bg-blue-500 transition-colors"
      :class="
        searchStore.sortType === SortType.ASCENDING
          ? 'bg-blue-500 text-white'
          : 'bg-transparent'
      "
      @click="onAscendingSort"
    >
      {{ SortType.ASCENDING }}
    </button>
    <button
      class="border border-blue-500 hover:text-white px-4 hover:bg-blue-500 transition-colors"
      :class="
        searchStore.sortType === SortType.DESCENDING
          ? 'bg-blue-500 text-white'
          : 'bg-transparent'
      "
      @click="onDescendingSort"
    >
      {{ SortType.DESCENDING }}
    </button>
  </div>
</template>
