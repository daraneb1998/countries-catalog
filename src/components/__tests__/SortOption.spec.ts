import { describe, it, expect, beforeEach } from "vitest";

import { mount } from "@vue/test-utils";
import { createPinia, setActivePinia } from "pinia";
import SortOption from "../SortOption.vue";

describe("SortOption", () => {
  beforeEach(() => {
    setActivePinia(createPinia());
  });

  it("renders properly", () => {
    const wrapper = mount(SortOption);
    expect(wrapper.text()).toMatchSnapshot();
  });
});
