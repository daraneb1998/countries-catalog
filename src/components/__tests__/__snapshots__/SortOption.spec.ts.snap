// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`SearchBox > renders properly 1`] = `""`;

exports[`SortOption > renders properly 1`] = `"SORT NAMEASCENDINGDESCENDING"`;
